src/subscriber.ts

```diff
@@ -209,27 +214,40 @@ export class Message {
   }
 }
 
-export interface SubscriberOptions {
-  ackDeadline?: number;
-  batching?: BatchOptions;
-  flowControl?: FlowControlOptions;
-  useLegacyFlowControl?: boolean;
-  streamingOptions?: MessageStreamOptions;
-  enableOpenTelemetryTracing?: boolean;
-}
-
 /**
  * @typedef {object} SubscriberOptions
  * @property {number} [ackDeadline=10] Acknowledge deadline in seconds. If left
- *     unset the initial value will be 10 seconds, but it will evolve into the
- *     99th percentile time it takes to acknowledge a message.
+ *     unset, the initial value will be 10 seconds, but it will evolve into the
+ *     99th percentile time it takes to acknowledge a message, subject to the
+ *     limitations of minAckDeadline and maxAckDeadline. If ackDeadline is set
+ *     by the user, then the min/max values will be set to match it. New code
+ *     should prefer setting minAckDeadline and maxAckDeadline directly.
+ * @property {Duration} [minAckDeadline] The minimum time that ackDeadline should
+ *     ever have, while it's under library control.
+ * @property {Duration} [maxAckDeadline] The maximum time that ackDeadline should
+ *     ever have, while it's under library control.
  * @property {BatchOptions} [batching] Request batching options.
  * @property {FlowControlOptions} [flowControl] Flow control options.
  * @property {boolean} [useLegacyFlowControl] Disables enforcing flow control
  *     settings at the Cloud PubSub server and uses the less accurate method
  *     of only enforcing flow control at the client side.
  * @property {MessageStreamOptions} [streamingOptions] Streaming options.
  */
+export interface SubscriberOptions {
+  /** @deprecated Use minAckDeadline and maxAckDeadline. */
+  ackDeadline?: number;
```

Comment:
What is the difference between `ackDeadline` and `minAckDeadline`/`maxAckDeadline`? If the user wanted an explicit `ackDeadline` would that be the same as `minAckDeadline = ackDeadline` and `maxAckDeadline = ackDeadline`?

Reply:
Yes. Setting ackDeadline will also set min/maxAckDeadline, to match the previous behaviour.